dynatrace/build.gradle

```diff
@@ -26,10 +26,13 @@ repositories {
     mavenCentral()
 }
 
-def otelVersion = '1.14.0'
-
 dependencies {
-    implementation platform("io.opentelemetry:opentelemetry-bom:${otelVersion}")
+    implementation platform("io.opentelemetry:opentelemetry-bom") {
+        version {
+            strictly '[1.14.0, 2.0['
+            prefer '1.20.1'
+        }
+    }
     implementation('io.opentelemetry:opentelemetry-sdk')
     implementation('io.opentelemetry:opentelemetry-sdk-metrics')
```

Comment:
```
    compileOnly platform("io.opentelemetry:opentelemetry-bom:1.14.0")
    compileOnly('io.opentelemetry:opentelemetry-sdk')
    compileOnly('io.opentelemetry:opentelemetry-sdk-metrics')
    testImplementation('io.opentelemetry:opentelemetry-sdk-testing:1.20.1')
```
Would mean: compiling our package against the 1.14.0 API, everything above should be fine as long as its backward compatible. We then can specify which version to run the tests against (so in the above case the version running in the tests is 1.20.1).

Reply:
Actually, I did the following test: 
Set the dependencies in dynatrace/build.gradle to
```
    implementation platform("io.opentelemetry:opentelemetry-bom") {
        version {
            strictly '[1.16.0, 2.0['
            prefer '1.20.1'
        }
    }
    implementation('io.opentelemetry:opentelemetry-sdk')
    implementation('io.opentelemetry:opentelemetry-sdk-metrics')
```
(Note the 1.16.0 instead of 1.14.0)
and the dependencies in example/build.gradle to
```
    implementation platform("io.opentelemetry:opentelemetry-bom:1.15.0!!")
    implementation("io.opentelemetry:opentelemetry-api")
    implementation("io.opentelemetry:opentelemetry-sdk")
```
so, lower than the supposed minimum version in the lib. It still works. If you use `implementation` in the dynatrace/build.gradle it will supply the 1.15 version as the runtime version also to the lib, if you use `compileOnly` the lib does not have a runtime dependency on the OTel packages (which seems to be fine). If you add the dep to the example app, everything works.